src/Microsoft.AspNet.OData.Shared/Query/Expressions/ExpressionBinderBase.cs

```diff
@@ -69,20 +69,7 @@ public abstract class ExpressionBinderBase
         /// </summary>
         internal IDictionary<string, Expression> FlattenedPropertyContainer;
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="ExpressionBinderBase"/> class.
-        /// </summary>
-        /// <param name="requestContainer">The request container.</param>
-        protected ExpressionBinderBase(IServiceProvider requestContainer)
-        {
-            Contract.Assert(requestContainer != null);
-
-            QuerySettings = requestContainer.GetRequiredService<ODataQuerySettings>();
-            Model = requestContainer.GetRequiredService<IEdmModel>();
-            AssembliesResolver = requestContainer.GetRequiredService<IAssembliesResolver>();
-        }
```

Comment:
you aren't sealing the class you are removing the protected ctor. does the new use case you are proposing work?

Reply:
The ctor is added back in Step4 on the AspNet side.